postgres/pgdump.go

```diff
@@ -470,18 +474,52 @@ func extractConstraints(conv *internal.Conv, n nodes.Node, table string, l []nod
 	for _, i := range l {
 		switch d := i.(type) {
 		case nodes.Constraint:
-			var cols []string
-			for _, j := range d.Keys.Items {
-				k, err := getString(j)
+			var cols, referCols []string
+			var referTable string
+			switch d.Contype {
+			case nodes.CONSTR_FOREIGN:
+				t, err := getTableName(conv, *d.Pktable)
 				if err == nil {
```

Comment:
Probably simpler to put this code after the error handling case.
And then the if isn't needed because of the continue.

Reply:
oh but how would I put error statement above case statement? It's only true for foreign key constraint and not for indexes(which we are going to do a PR) and for primary keys.

Reply:
Moved referTable = t statement after error check block